src/Database/Expression/GroupedExpression.php

```diff
@@ -0,0 +1,113 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.1.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database\Expression;
+
+use Cake\Database\ValueBinder;
+use Closure;
+
+/**
+ * Handles the few special aggregate functions that require special syntax
+ * for grouped ordering.
+ */
+class GroupedExpression extends FunctionExpression
+{
+    /**
+     * @var \Cake\Database\Expression\FunctionExpression
+     */
+    protected $orderContainer;
+
+    /**
+     * @var \Cake\Database\Expression\OrderByExpression|null
+     */
+    protected $order;
+
+    /**
+     * @inheritDoc
+     */
+    public function __construct(string $name, array $params = [], array $types = [], string $returnType = 'string')
```

Comment:
If this expression is only used for GroupConcat does the name need to be a parameter?